duckduckgo/Android
# Suppress download dialog if user already confirmed download with a different dialog

Task/Issue URL: https://app.asana.com/0/414730916066338/1170085036962994

**Description**:
PR https://github.com/duckduckgo/Android/pull/663 added a new download confirmation dialog (currently merged into a feature branch). This PR suppresses that dialog in cases where the user has already interacted with a dialog e.g accepted the Android system permissions dialog or long pressed and selected "download image".

**Steps to test this PR**:
1. Perform a **fresh** installation of the app
1. Go to the serp and search for "test"
1. Tap on "images"
1. Long press an image and select "download image"
1. Note that you are shown the system permissions dialog
1. Grant permission and note you are **not** shown the download confirmation dialog
1. Long press another image and select "download image" again
1. Note you are **not** shown the download confirmation dialog
1. Go to  https://github.com/duckduckgo/Android/releases and download an apk
1. Note you **are** shown the download confirmation dialog this time

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
